packages/voting-contract/src/lib.rs

```diff
@@ -271,6 +271,34 @@ where
     Ok(ProposalListResponse { proposals: props? })
 }
 
+pub fn list_text_proposals<P>(
+    deps: Deps,
+    env: Env,
+    start_after: Option<u64>,
+    limit: usize,
+) -> StdResult<ProposalListResponse<P>>
+where
+    P: Serialize + DeserializeOwned + TextProposal,
+{
+    let start = start_after.map(Bound::exclusive_int);
+    let props: StdResult<Vec<_>> = proposals()
+        .range(deps.storage, start, None, Order::Ascending)
+        .filter(
+            |r: &Result<(u64, state::Proposal<P>), cosmwasm_std::StdError>| {
```

Comment:
That's interesting, I was kind of surprised that compiler required explicit type.
Even though compiler's suggestion here is 100% on point - it literally advices to use that type:
```
287 |             |r| {
    |              ^ consider giving this closure parameter the
explicit type `&Result<(u64, state::Proposal<P>), cosmwasm_std::StdError>`,
where the type parameter `P` is specified
```
I assume that every time generic parameter is present, closure needs to be explicit?

Reply:
I think so? I didn't investigate to be honest, just did what the compiler suggested. It does feel redundant.